### PR TITLE
Remove unneded helper

### PR DESCRIPTION
**Pull request by potuz**
Remove unneeded check and consequently simplify the helper.

### CODE REVIEW

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -528,23 +527,6 @@ func (s *Service) handleEpochBoundary(ctx context.Context, postState state.Beaco
 	return nil
 }
 
-// This feeds in the block to fork choice store. It's allows fork choice store
-// to gain information on the most current chain.
-func (s *Service) insertBlockToForkchoiceStore(ctx context.Context, blk interfaces.ReadOnlyBeaconBlock, root [32]byte, st state.BeaconState) error {
-	ctx, span := trace.StartSpan(ctx, "blockChain.insertBlockToForkchoiceStore")
-	defer span.End()
-
-	if !s.cfg.ForkChoiceStore.HasNode(blk.ParentRoot()) {
```
Comment (potuz) [inline beacon-chain/blockchain/process_block.go:537]: This check is performed at the beginning of `onBlock`

Comment (terencechain) [reply beacon-chain/blockchain/process_block.go:537]: It's different though, `onBlock` doesn't call `fillInForkChoiceMissingBlocks`. Only `onBlockBatch` does

Comment (potuz) [reply beacon-chain/blockchain/process_block.go:537]: I didn't remove that helper

Comment (terencechain) [reply beacon-chain/blockchain/process_block.go:537]: My bad. I missed the "can't be triggered" part

**Review by terencechain — APPROVED** [review]

**Pull request merged**